mloop/learners.py

```diff
@@ -1762,8 +1763,9 @@ def __init__(self,
         else:
             self.noise_level = float(noise_level)
         self.update_hyperparameters = bool(update_hyperparameters)
+        # Make the length scale bounds default on the order of our scaled parameters (order ~1)
         if length_scale_bounds is None:
-            self.length_scale_bounds = np.array([1e-5, 1e5])
+            self.length_scale_bounds = np.array([1e-3, 1e1])
```

Comment:
I think we need to handle the length scales a bit differently.

We should make it so that user doesn't really need to be aware that many values, including parameter values, are rescaled behind-the-scenes. With that goal in mind, the length scales specified by the user should be in real units, then we'll scale those values to scaled units when necessary. In particular, we'll want to use the scaled length scales in `GaussianProcessLearner.create_gaussian_process()`. Keep in mind that we actually don't want to use `GaussianProcessLearner.params_scaler.transform()` for this because we don't want to offset the length scales - we just want to scale them by the right multiplicative factor. I believe this can be done by multiplying the length scales and their bounds by `GaussianProcessLearner.params_scaler.scale_`. We'll also need to unscale the fitted value by dividing the fitted value by `GaussianProcessLearner.params_scaler.scale_` before storing the result as `self.length_scale` and in `self.length_scale_history` during `GaussianProcessLearner.fit_gaussian_process()`. To keep things nicely organized, readable, and easy-to-maintain, let's add two simple methods, `GaussianLearner._transform_length_scales()` and `GaussianLearner._inverse_transform_length_scales()` which perform that multiplication/division.

These changes are also necessary for backwards compatibility. Previously the user specified length scales in real units. But with the PR in its current state, the parameter values are scaled but the length scales aren't, so now the length scales effectively have to be specified in scaled units. We need to scale the length scales as well. That way the length scale arguments provided by the user are interpreted as being in real units, just as they were before this PR.

One slight caveat to the "the user shouldn't know about scaled units" point is that I think the default length scale values should effectively be in scaled units. This is necessary for us to give it reasonable values no matter whether the user's parameter has a range ~`1e10` or ~`1e-10`. We can express this to the user by saying something like "By default, the length scale for each parameter is bounded to be between 0.001 to 10 times the allowed range for that parameter" in the docstring. We'll say something along those lines for the default length scale value as well. That explanation informs the user how the default value is set without having to introduce them to the concept of scaled units, even though we're really just telling them the default length scale bounds in scaled units.

Implementing those default values will take some slight reorganization in `GaussianLearner.__init__()`. We'll need to initialize the `ParameterScaler` and run its `partial_fit()` method slightly earlier on so that its `scale_` attribute is available when we set the default values for `self.length_scale` and `self.length_scale_bounds`.

Reply:
I've seen this and that all makes sense. I'll try to get to this soon. Thanks for the feedback!

Reply:
Hey @matth2peters I actually want to take advantage of the changes in this PR soon, so if you don't mind I'd like to add a few commits to this PR to wrap up this last issue

Reply:
Yeah of course! Sorry, I haven’t forgotten, I just haven’t had time to revisit this yet since the semester started. Feel free to implement this!